stellargraph/layer/graph_classification.py

```diff
@@ -250,3 +243,123 @@ def in_out_tensors(self):
         return x_inp, x_out
 
     build = deprecated_model_function(in_out_tensors, "build")
+
+
+@experimental(reason="Missing unit tests and generally untested.")
+class DeepGraphConvolutionalNeuralNetwork(GCNSupervisedGraphClassification):
```

Comment:
On the topic of naming (separate comment, because it's orthogonal to the code-sharing discussion above), we have `GCN` not `GraphConvolutionalNetwork` and `GAT` not `GraphAttentionNetwork` etc. Thus, maybe this should be:

```suggestion
class DGCNN(GCNSupervisedGraphClassification):
```

Reply:
Main reason I went with the full name was that I felt we are oversaturated with acronyms and thought we might want to start naming classes using the full name of the algorithms. 

I leave it up to you to decide which is best.